book/listings/actions/1/main.rs

```diff
@@ -31,11 +30,12 @@ fn build_ui(app: &Application) {
         .build();
 
     // Add action "close" to `window` taking no parameter
-    let action_close = SimpleAction::new("close", None);
-    action_close.connect_activate(clone!(@weak window => move |_, _| {
-        window.close();
-    }));
-    window.add_action(&action_close);
+    let action_close = ActionEntry::builder("close")
+        .activate(|window: &ApplicationWindow, _, _| {
+            window.close();
+        })
+        .build();
+    window.add_action_entries([action_close]);
```

Comment:
The action exists already?

Reply:
You mean `window.close`?
Yes, I mention that in the chapter.

Reply:
ah never mind me then :)